packages/itwinui-react/src/utils/providers/ScopeProvider.tsx

```diff
@@ -11,14 +11,18 @@ const ScopeContext = React.createContext({
   parentStore: null as null | ReturnType<typeof createStore>,
 });
 
+type ScopeProviderProps = { children: React.ReactNode };
```

Comment:
I'm not sure how helpful it is to have these `type`s defined separately. It adds a lot of unnecessary top-level noise to the file.

In private components, we often just have one or two props. And in public components, we usually need `PolymorphicForwardRefComponent`. So I'm thinking that we have outgrown the convention defined in `STYLEGUIDE.md`.

Reply:
The `STYLEGUIDE` is meant to be an evolving document based on the team's (current) agreed-upon preferences. Many points in it were written quite some time ago, so when in doubt, you can always ask internally before changing code.